# Whitney Hardy 
## Task 4.3 Intro to Pandas

### Content List
#### Import libraires
#### Make path 
#### Orders 
#### Checking prices of Products- Changed milk and cottage cheese price to be more appropriate
#### Answer the following questions using relevant commands


### Import Libraires

In [1]:
# Import libraries
import pandas as pd
import numpy as np
import os

### make path

In [2]:
path = r'C:\Users\hardy\10-2025 Instacart Basket Analysis'

In [3]:
path

'C:\\Users\\hardy\\10-2025 Instacart Basket Analysis'

### Orders

In [4]:
df = pd.read_csv(r'C:\Users\hardy\10-2025 Instacart Basket Analysis\02 Data\Original Data\orders.csv', index_col = False) 

In [5]:
df = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'orders.csv'), index_col = False)

In [6]:
vars_list = ['order_id', 'user_id', 'order_number', 'order_dow', 'order_hour_of_day', 'days_since_prior_order'] 

In [7]:
vars_list

['order_id',
 'user_id',
 'order_number',
 'order_dow',
 'order_hour_of_day',
 'days_since_prior_order']

In [8]:
#print first 5 rows
df.head(5) 

order_id  user_id eval_set  order_number  order_dow  order_hour_of_day  \
0   2539329        1    prior             1          2                  8   
1   2398795        1    prior             2          3                  7   
2    473747        1    prior             3          3                 12   
3   2254736        1    prior             4          4                  7   
4    431534        1    prior             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

### Products

In [9]:
df_prods = pd.read_csv(os.path.join(path, '02 data', 'Original Data', 'products.csv'), index_col = False)  

In [10]:
df_prods.head(20) 

product_id                                       product_name  aisle_id  \
0            1                         Chocolate Sandwich Cookies        61   
1            2                                   All-Seasons Salt       104   
2            3               Robust Golden Unsweetened Oolong Tea        94   
3            4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4            5                          Green Chile Anytime Sauce         5   
5            6                                       Dry Nose Oil        11   
6            7                     Pure Coconut Water With Orange        98   
7            8                  Cut Russet Potatoes Steam N' Mash       116   
8            9                  Light Strawberry Blueberry Yogurt       120   
9           10     Sparkling Orange Juice & Prickly Pear Beverage       115   
10          11                                  Peach Mango Juice        31   
11          12                         Chocolate Fudge Layer Cake       119   
12          13                                  Saline Nasal Mist        11   
13          14                     Fresh Scent Dishwasher Cleaner        74   
14          15                           Overnight Diapers Size 6        56   
15          16                      Mint Chocolate Flavored Syrup       103   
16          17                                  Rendered Duck Fat        35   
17          18                Pizza for One Suprema  Frozen Pizza        79   
18          19   Gluten Free Quinoa Three Cheese & Mushroom Blend        63   
19          20     Pomegranate Cranberry & Aloe Vera Enrich Drink        98   

    department_id  prices  
0              19     5.8  
1              13     9.3  
2               7     4.5  
3               1    10.5  
4              13     4.3  
5              11     2.6  
6               7     4.4  
7               1     1.1  
8              16     7.0  
9               7     8.4  
10              7     2.8  
11              1     9.4  
12             11     1.1  
13             17     6.5  
14             18    11.2  
15             19     5.2  
16             12    17.1  
17              1    10.6  
18              9    12.0  
19              7     6.0

In [11]:
df_prods.tail(35) 

product_id                                       product_name  \
49658       49654                   Teriyaki Sauce, Sesame, Original   
49659       49655                                        Apple Cider   
49660       49656                         Masada Kosher Pocket Bread   
49661       49657                                  Cabernet Tomatoes   
49662       49658                         Brie with Herbs Foil Wedge   
49663       49659                            Organic Creamed Coconut   
49664       49660                        Professionals Sleek Shampoo   
49665       49661                                              Porto   
49666       49662                       Bacon Cheddar Pretzel Pieces   
49667       49663  Ultra Protein Power Crunch Peanut Butter N' Ho...   
49668       49664                     Lemon Cayenne Drinking Vinegar   
49669       49665      Super Dark Coconut Ash & Banana Chocolate Bar   
49670       49666                      Ginger Snaps Snacking Cookies   
49671       49667           Enchilada with Spanish Rice & Beans Meal   
49672       49668                     Apple Cinnamon Scented Candles   
49673       49669                                   K Cup Dark Blend   
49674       49670                                Beef Summer Sausage   
49675       49671                               Milk Chocolate Drops   
49676       49672                             Cafe Mocha K-Cup Packs   
49677       49673              Stone Baked Multi Grain Artisan Rolls   
49678       49674            Frozen Greek Yogurt Bars Chocolate Chip   
49679       49675                Cinnamon Dolce Keurig Brewed K Cups   
49680       49676                             Ultra Red Energy Drink   
49681       49677                    Thick & Chunky Sloppy Joe Sauce   
49682       49678                    Large Chicken & Cheese Taquitos   
49683       49679                            Famous Chocolate Wafers   
49684       49680                 All Natural Creamy Caesar Dressing   
49685       49681            Spaghetti with Meatballs and Sauce Meal   
49686       49682                                 California Limeade   
49687       49683                                     Cucumber Kirby   
49688       49684          Vodka, Triple Distilled, Twist of Vanilla   
49689       49685                 En Croute Roast Hazelnut Cranberry   
49690       49686                                   Artisan Baguette   
49691       49687         Smartblend Healthy Metabolism Dry Cat Food   
49692       49688                             Fresh Foaming Cleanser   

       aisle_id  department_id  prices  
49658         5             13     4.0  
49659        98              7    10.7  
49660       128              3     7.1  
49661        83              4     8.3  
49662         2             16     3.9  
49663        17             13     3.1  
49664        22             11     6.7  
49665       134              5     8.2  
49666       107             19     3.6  
49667        57             14    10.2  
49668       100             21    13.7  
49669        45             19     6.9  
49670        61             19     5.2  
49671        38              1     6.6  
49672       101             17     5.6  
49673       100             21     4.7  
49674       106             12    19.2  
49675        45             19     3.0  
49676        26              7     6.5  
49677       129              1     5.6  
49678        37              1    11.1  
49679        26              7    14.0  
49680        64              7    14.5  
49681        59             15     8.9  
49682       129              1     3.4  
49683        61             19     6.0  
49684        89             13     4.9  
49685        38              1     6.9  
49686        98              7     4.3  
49687        83              4    13.2  
49688       124              5     5.3  
49689        42              1     3.1  
49690       112              3     7.8  
49691        41     

#### Checking prices of Products- Changed milk and cottage cheese price to be more appropriate 

In [18]:
#check prices
top_50_expensive = df_prods.nlargest(50, 'prices') 

In [13]:
top_50_expensive 

product_id                                       product_name  \
33666       33664                              2 % Reduced Fat  Milk   
21554       21553                   Lowfat 2% Milkfat Cottage Cheese   
9020         9020                   Boneless Skinless Chicken Thighs   
19392       19391                          Turkey Breast Tenderloins   
21468       21467                             Wild Caught Raw Shrimp   
25580       25579                      Naturally Smoked Trout Fillet   
40490       40486                                    Chicken Tenders   
9896         9896                     Uncured Applewood Smoked Bacon   
14207       14207                                   Angus Roast Beef   
36577       36573                                     Peppered Bacon   
39050       39046                            Smok Cured Turkey Bacon   
41097       41093                           Sugar Free Dry Rub Bacon   
48343       48339                                      Lobster Tails   
9302         9302                         Thin Sliced Chicken Breast   
19548       19547                                  Beef Frankfurters   
22193       22192                                   Lite Beef Franks   
26078       26077                                  Cocktail Sausages   
33987       33985                                   Sausage, Chorizo   
10814       10814                             Applewood Smoked Bacon   
21141       21140    Grillers Cheddar Cheese & Bacon Patties 6 Count   
22956       22955                                          Diced Ham   
30545       30543                              Herring In Wine Sauce   
38752       38748                                    Polish Kielbasa   
8441         8441                 Farmland Hickory Smoked Bacon 24oz   
23806       23805                              Peppered Turkey Bacon   
24884       24883                             Bone In Chicken Thighs   
38950       38946                     Everyday Original Turkey Bacon   
1992         1992                       Beef (101445) Summer Sausage   
27484       27482                                     Turkey Sausage   
44570       44566                                    Thick Cut Bacon   
45644       45640                               Lite Polska Kielbasa   
47884       47880                           Turkey Cranberry Sausage   
8914         8914                                     Chicken Breast   
16972       16972                               Fresh Catfish Fillet   
19001       19000                           Wild Caught Sea Scallops   
20124       20123                                  Smoked Ham Shanks   
47786       47782                                     Lamb Stew Meat   
83             84                                         Lamb Shank   
3529         3529                        Fully Cooked Original Bacon   
22071       22070     Fully Cooked Breakfast Sausage Original Recipe   
24087       24086                              Beef Tenderloin Steak   
46390       46386                                Ground Sirloin Beef   
5646         5646                               Organic Turkey Bacon   
8743         8743                Fully Cooked Turkey Sausage Patties   
7337         7337                        Supremo Pork Chorizo 12 Oz.   
10197       10197           Chicken Meatballs Spicy Mango & Jalapeno   
14453       14453                                      Uncured Bacon   
41675       41671          Original Patties (100965) 12 Oz Breakfast   
41880       41876  Premium Hardwood Smoked Seasoned Angus Beef Fr...   
12841       12841                                     King Crab Legs   

       aisle_id  department_id   prices  
33666        84             16  99999.0  
21554       108             16  14900.0  
9020         35             12     25.0  
19392        49             12     25.0  
21468        15             12     25.0  
25580        15             12     25.0  
40490        49             12     25.0 

In [14]:
# Filter for department 16 and select specific columns
df_prods[df_prods['department_id'] == 16][['product_id', 'product_name', 'prices']] 

product_id                                       product_name  prices
8               9                  Light Strawberry Blueberry Yogurt     7.0
97             98                          2% Yellow American Cheese     4.5
116           117                                 Petit Suisse Fruit     3.1
120           121                                      Sharp Cheddar     8.1
126           127                                         Marscapone     3.5
...           ...                                                ...     ...
49621       49617  Creamy Mozzarella Sun-Dried Tomato & Basil Fla...    10.2
49626       49622                     Original Powder Coffee Creamer    12.0
49632       49628                                  Yoghurt Blueberry    14.7
49648       49644                                      Feta Crumbles     6.5
49662       49658                         Brie with Herbs Foil Wedge     3.9

[3449 rows x 3 columns]

In [15]:
# Change price for product_id 33664 (2% Reduced Fat Milk)
df_prods.loc[df_prods['product_id'] == 33664, 'prices'] = 9.99

In [16]:
# Change price for product_id 21553 (Lowfat 2% Milkfat Cottage Cheese)
df_prods.loc[df_prods['product_id'] == 21553, 'prices'] = 14.90

In [17]:
# Verify the changes
df_prods[df_prods['product_id'].isin([33664, 21553])][['product_id', 'product_name', 'prices']] 

product_id                      product_name  prices
21554       21553  Lowfat 2% Milkfat Cottage Cheese   14.90
33666       33664             2 % Reduced Fat  Milk    9.99

#### Returning back to task steps

In [19]:
#Print the names of the columns in the df_prods dataframe 
df_prods.columns 

Index(['product_id', 'product_name', 'aisle_id', 'department_id', 'prices'], dtype='object')

In [20]:
#print number of rows and columns 
df_prods.shape

(49693, 5)

#### Answer the following questions using relevant commands 

In [21]:
# What's the maximum value of the “aisle_id” column?
df_prods.describe() 

product_id      aisle_id  department_id        prices
count  49693.000000  49693.000000   49693.000000  49693.000000
mean   24844.345139     67.770249      11.728433      7.682460
std    14343.717401     38.316774       5.850282      4.200295
min        1.000000      1.000000       1.000000      1.000000
25%    12423.000000     35.000000       7.000000      4.100000
50%    24845.000000     69.000000      13.000000      7.100000
75%    37265.000000    100.000000      17.000000     11.200000
max    49688.000000    134.000000      21.000000     25.000000

The max value of the aisle_id column is 134

In [22]:
#What’s the data type of the “department_id” column?
df_prods.dtypes

product_id         int64
product_name      object
aisle_id           int64
department_id      int64
prices           float64
dtype: object

Depanrtment_id is an integer data type.